src/main/java/ru/svgogin/service/spark/repository/Repository.java

```diff
@@ -0,0 +1,88 @@
+package ru.svgogin.service.spark.repository;
+
+import java.time.LocalDate;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.PostConstruct;
+import ru.svgogin.service.spark.dto.CompanyDto;
+import ru.svgogin.service.spark.entity.Company;
+
+@org.springframework.stereotype.Repository
+public class Repository {
+  private final Map<String, Company> companyMap = new LinkedHashMap<>();
+
+  @PostConstruct
+  public void init() {
+    save(new CompanyDto("7729355614",
+        "1027700262270",
+        "770401001",
+        "АКЦИОНЕРНОЕ ОБЩЕСТВО \"ДОМ.РФ\"",
+        "АО\"ДОМ.РФ\"",
+        "Действующая",
+        LocalDate.of(2020, 12, 31)));
+
+    save(new CompanyDto("7725038124",
+        "1037739527077",
+        "770401001",
+        "АКЦИОНЕРНОЕ ОБЩЕСТВО \"БАНК ДОМ.РФ\"",
+        "АО\"БАНК ДОМ.РФ\"",
+        "Действующая",
+        LocalDate.of(2020, 11, 30)));
+  }
+
+  public Iterable<CompanyDto> findAll() {
+    return companyMap.values().stream()
+        .map(this::toDto)
+        .collect(Collectors.toList());
+  }
+
+  public void save(CompanyDto companyDto) {
+    if (companyMap.containsKey(companyDto.getInn())) {
+      throw new IllegalArgumentException("Company inn = " + companyDto.getInn()
+                                         + " already exists");
+    }
+    companyMap.put(companyDto.getInn(), toEntity(companyDto));
+  }
+
+  public CompanyDto findByInn(String inn) {
+    return toDto(companyMap.get(inn));
+  }
+
+  private Company toEntity(CompanyDto companyDto) {
+    return new Company(companyDto.getInn(),
+        companyDto.getOgrn(),
+        companyDto.getKpp(),
+        companyDto.getFullNameRus(),
+        companyDto.getShortNameRus(),
+        companyDto.getStatusName(),
+        companyDto.getStatusDate());
+  }
+
+  private CompanyDto toDto(Company company) {
+    return new CompanyDto(company.getInn(),
+        company.getOgrn(),
+        company.getKpp(),
+        company.getFullNameRus(),
+        company.getShortNameRus(),
+        company.getStatusName(),
+        company.getStatusDate());
+  }
```

Comment:
it is better to convert entity to dto and dto to entity in service and repository should work only with entities